src/context/UserContext.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  createContext,
+  useContext,
+  useReducer,
+  ReactNode,
+  Dispatch,
+} from 'react';
+
+interface UserState {
+  name: string;
+  email: string;
+  isLoggedIn: boolean;
+  currency: string;
+  language: string;
+  theme: string;
```

Comment:
We only support light and dark there. There could be more but it's a limited list with values that we know. It's better to use `enum` here as well.

Reply:
I added enum for Light and Dark.